model/flow/sealing_segment.go

```diff
@@ -94,6 +103,11 @@ func (builder *SealingSegmentBuilder) SealingSegment() (*SealingSegment, error)
 
 	// if root sealing segment skip seal sanity check
 	if len(segment.Blocks) == 1 {
+
+		if !builder.isValidRootSegment() {
+			return nil, fmt.Errorf("root sealing segment block has the wrong view got (%d) expected (%d): %w", segment.Highest().Header.View, rootSegmentBlockView, ErrInvalidRootSegmentView)
```

Comment:
Here we are decoupling the error context (line 108) from the logic which defines the conditions that produce the error (lines 168-170), which makes it more difficult to verify that the logic and error context remain consistent.

What do you think about using a pattern of returning errors from validation functions, so that the error context from the validation function's logic can live within the validation function.
```go
func validateRootSegment() error {
   if view != expectedView {
     return fmt.Errorf("wrong view......") 
   }
   // ...
}

// ...
if err := builder.validateRootSegment(); err != nil {
  return nil, fmt.Errorf("could not validate root segment: %w", err)
}
```

Reply:
I like this thanks

Reply:
https://github.com/onflow/flow-go/pull/1533/commits/ef3887b8355cf2913aa1911d598883538769238a